Core/Graphics/Source/BgfxCallback.cpp

```diff
@@ -123,15 +125,28 @@ namespace Babylon
         });
     }
 
-    void BgfxCallback::captureBegin(uint32_t /*width*/, uint32_t /*height*/, uint32_t /*pitch*/, bgfx::TextureFormat::Enum /*format*/, bool /*yflip*/)
+    void BgfxCallback::captureBegin(uint32_t width, uint32_t height, uint32_t pitch, bgfx::TextureFormat::Enum format, bool yflip)
     {
+        m_captureData.Width = width;
+        m_captureData.Height = height;
+        m_captureData.Pitch = pitch;
+        m_captureData.Format = format;
+        m_captureData.YFlip = yflip;
     }
 
     void BgfxCallback::captureEnd()
     {
     }
 
-    void BgfxCallback::captureFrame(const void* /*_data*/, uint32_t /*_size*/)
+    void BgfxCallback::captureFrame(const void* data, uint32_t size)
     {
+        m_captureData.Data = data;
+        m_captureData.DataSize = size;
+
+        std::scoped_lock lock{m_captureCallbacksMutex};
+        for (const auto& callback : m_captureCallbacks)
+        {
+            callback(m_captureData);
+        }
```

Comment:
Might be good to reset m_captureData to defaults here to avoid confusion. Maybe do it in captureEnd?

Reply:
Maybe don't store m_captureData at all and just store each individual width, height, pitch, etc. instead.

Reply:
I'm not sure I understand. Can you elaborate on what confusion these actions might help to avoid?